---
title: "Supplementary Material - Using high-resolution variant frequencies to empower clinical genome interpretation"
output: 
  word_document:
    reference_docx: template2.docx    

bibliography: "suppReferences.bib"
csl: "nature.csl"
---

```{r, echo=FALSE, include=FALSE}
#read all scripts in the R directory, to load functions
sourceDir <- function(path, trace = TRUE, ...) {
    for (nm in list.files(path, pattern = "\\.[RrSsQq]$")) {
       if(trace) cat(nm,":")           
       source(file.path(path, nm), ...)
       if(trace) cat("\n")
    }
}

#load functions
#sourceDir(file.path("..","R"))
source("../R/setup.R")
```

### Supplementary table 1

Variants previously reported as causative of HCM either in ClinVar, or in a clinical series of 6179 HCM cases, but that were observed in ExAC above the maximum tolerated allele count for HCM (AC>9 globally) were manually curated according to the ACMG guidelines for interpretation and classification on sequence variants. *These variants were previously categorised as Likely Pathogenic in the LMM and Oxford HCM cohorts.

```{r}
read.delim("../data-raw/SupplementaryTable1.txt") %>% kable
```

### Supplementary note 1 - Curation of a high frequency PCD variant

_NME8_ NM\_016616.4:c.271-27C>T which is reported as Pathogenic in ClinVar is found in 2306/120984 ExAC individuals. This variant was initially reported as pathogenic on the basis of two compound heterozygous cases when specifically searching for NME8 variants in a set of patients, and was found in 2/196 control chromosomes `r citep("10.1073/pnas.0611405104")`. We further note that _NME8_ has not otherwise been associated with PCD and shows no evidence of missense nor loss-of-function constraint and that this splice variant affects a non-canonical transcript. During our curation exercise we found that this variant meets none of the ACMG criteria for assertions of pathogenicity, and therefore we reclassified it as a VUS. 

### Supplementary note 2 - Dealing with penetrance

It is often difficult to obtain accurate penetrance information for reported variants, and it is also difficult to know what degree of penetrance to expect or assume in newly discovered pathogenic variants. In this work we uniformly apply a value of 50% penetrance equivalent to that reported for our HCM example variant, and the lowest we found reported for any of our examples. We recognise several other approaches that can be used to deal with the issues of penetrance, these include: setting a penetrance level equivalent to the minimum that is 'clinically actionable' for a disorder; lowering the penetrance if reduced penetrance is expected in a family; or using a two tiered approach, initially searching for a high-moderate penetrance variant but allowing for a lower-penetrance variant in a second pass. We believe that the ease of re-calculating our "maximum credible population allele frequency" lends itself to any of these approaches. We provide an online calculator to facilitate the exploration of these parameters (<http://139.59.161.186/> *temporary URL for review - username/password = frequency/filter*).

##Supplemental references

```{r, echo=FALSE, message=FALSE}
write.bibtex(file="suppReferences.bib")
```
